src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+
+            return new OnnxTransform(env, args, modelBytes);
+        }
+
+        // Factory method for SignatureLoadRowMapper.
+        public static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes = null)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register(RegistrationName);
+            _host.CheckValue(args, nameof(args));
+            _host.CheckNonWhiteSpace(args.InputColumn, nameof(args.InputColumn));
+            _host.CheckNonWhiteSpace(args.OutputColumn, nameof(args.OutputColumn));
+
+            if (modelBytes == null)
+            {
+                _host.CheckNonWhiteSpace(args.ModelFile, nameof(args.ModelFile));
+                _host.CheckUserArg(File.Exists(args.ModelFile), nameof(args.ModelFile));
+                _model = new OnnxModel(args.ModelFile);
+            }
+            else
+                _model = OnnxModel.CreateFromBytes(modelBytes);
+
+            Inputs = new[] { args.InputColumn };
+            Outputs = new[] { args.OutputColumn };
+            _args = args;
+        }
+
+        public OnnxTransform(IHostEnvironment env, string modelFile, string inputColumn, string outputColumn)
+            : this(env, new Arguments() { ModelFile = modelFile, InputColumn = inputColumn, OutputColumn = outputColumn })
+        {
+        }
+
+        public ISchema GetOutputSchema(ISchema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            foreach (var input in Inputs)
+            {
+                if (!inputSchema.TryGetColumnIndex(input, out int srcCol))
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input);
+            }
+
+            var transform = Transform(new EmptyDataView(_host, inputSchema));
+            return transform.Schema;
+        }
+
+        private IRowMapper MakeRowMapper(ISchema schema) => new Mapper(_host, this, schema);
+
+        private RowToRowMapperTransform MakeDataTransform(IDataView input)
+        {
+            _host.CheckValue(input, nameof(input));
+            return new RowToRowMapperTransform(_host, input, MakeRowMapper(input.Schema));
+        }
+
+        public IDataView Transform(IDataView input) => MakeDataTransform(input);
+
+        public void Save(ModelSaveContext ctx)
+        {
+            _host.AssertValue(ctx);
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            ctx.SaveBinaryStream("OnnxModel", w => { w.WriteByteArray(_model.ToByteArray()); });
+            ctx.SaveNonEmptyString(_args.InputColumn);
+            ctx.SaveNonEmptyString(_args.OutputColumn);
+        }
+
+        internal sealed class Mapper : IRowMapper
```

Comment:
>internal [](start = 8, length = 8)

why not private?

Reply:
Made private

Reply:
made private

---
In reply to: [219613222](https://github.com/dotnet/machinelearning/pull/942#discussion_r219613222) [](ancestors = 219613222)